section2-the-ruby-programming-language/numbers/calculator.rb

```diff
@@ -0,0 +1,20 @@
+# prints the title
+puts "Simple calculator"
+# added a line using 25 times of "-"
+25.times { print "-"}
+# going to the next line
+print "\n"
+
+# asking for the first number input from user
+puts "Enter the first number:"
+num_1 = gets.chomp
+# asing for the second number input from the user
+puts "Enter the second number:"
+num_2 = gets.chomp
+
+# outputs the first number + - x / % by the second number
+puts "The first number added by the second number is #{num_1.to_f + num_2.to_f}"
+puts "The first number deducted by the second number is #{num_1.to_f - num_2.to_f}"
+puts "The first number multiplied by the second number is #{num_1.to_f * num_2.to_f}"
+puts "The first number divided by the second number is #{num_1.to_f / num_2.to_f}"
```

Comment:
Why do you need to do like this?
```
num_1.to_f / num_2.to_f
```

Reply:
In order for arithmetic operations to work, we need it to be an integer or float. `get.chomp` gets input as string object so by using `.to_f`, we changes the string object to float when using division `/`.

Reply:
Why not `.to_i` ?

Reply:
If using `.to_i` to convert to integer, when divided the output will be rounded up. Let's say `10.to_i / 8.to_i` will give us  `2`. Using the `.to_f` to convert to float, gives us an accurate answer compared to integer.

Reply:
what if 10/3?

Reply:
`10.to_i / 3.to_i` gives us `3`
`10.to_f / 3.to_f` gives us `3.3333333333333335`
`(10.to_f / 3.to_f).round(2)` gives us `3.33` in two decimal places

Reply:
```
10.to_i / 3.to_i gives us 3
```
why?

Reply:
Because the output was rounded up to an integer?

Reply:
10.to_i is the same as 10
3.to_i is the same as 3.

Logically (mathematically), 10/3 should be 3.333. Why is it resulting in 3 insteaD?

Reply:
Because it is doing an integer division when using both integers.

Reply:
Why?
That don't make sense in maths no?

Reply:
In integer division and modulus, the dividend is divided by the divisor into an integer quotient and a remainder. The integer quotient operation is referred to as integer division, and the integer remainder operation is the modulus. In the real world of data manipulation there are some things that are always handled in whole units or numbers (ex. integer data type). Thus when using two integers it results in an integer division.